flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/MaterializedTableStatementITCase.java

```diff
@@ -236,6 +252,213 @@ void testCreateMaterializedTableInFullMode() {
+                        + " PARTITIONED BY (ds)\n"
+                        + " WITH(\n"
+                        + "   'format' = 'debezium-json'\n"
+                        + " )\n"
+                        + " FRESHNESS = INTERVAL '2' SECOND\n"
+                        + " AS SELECT \n"
+                        + "  user_id,\n"
+                        + "  shop_id,\n"
+                        + "  ds,\n"
+                        + "  COUNT(order_id) AS order_cnt\n"
+                        + " FROM (\n"
+                        + "    SELECT user_id, shop_id, order_created_at AS ds, order_id FROM my_source"
+                        + " ) AS tmp\n"
+                        + " GROUP BY (user_id, shop_id, ds)";
+
+        OperationHandle materializedTableHandle =
+                service.executeStatement(
+                        sessionHandle, materializedTableDDL, -1, new Configuration());
+        awaitOperationTermination(service, sessionHandle, materializedTableHandle);
+
+        // verify data exists in materialized table
+        CommonTestUtils.waitUtil(
+                () ->
+                        fetchTableData(sessionHandle, "SELECT * FROM my_materialized_table").size()
+                                == data.size(),
+                Duration.ofMillis(timeout),
+                Duration.ofMillis(pause),
+                "Failed to verify the data in materialized table.");
+        assertThat(
+                        fetchTableData(
+                                        sessionHandle,
+                                        "SELECT * FROM my_materialized_table where ds = '2024-01-02'")
+                                .size())
+                .isEqualTo(2);
+
+        // remove the last element
+        data.remove(2);
+
+        long currentTime = System.currentTimeMillis();
+        String alterStatement =
+                "ALTER MATERIALIZED TABLE my_materialized_table REFRESH PARTITION (ds = '2024-01-02')";
+        OperationHandle alterHandle =
+                service.executeStatement(sessionHandle, alterStatement, -1, new Configuration());
+        awaitOperationTermination(service, sessionHandle, alterHandle);
+        List<RowData> result = fetchAllResults(service, sessionHandle, alterHandle);
+        assertThat(result.size()).isEqualTo(1);
+        String jobId = result.get(0).getString(0).toString();
+
+        MiniCluster miniCluster = MINI_CLUSTER.getMiniCluster();
+
+        // 1. verify a new job is created
+        Optional<JobStatusMessage> job =
+                miniCluster.listJobs().get(timeout, TimeUnit.MILLISECONDS).stream()
+                        .filter(j -> j.getJobId().toString().equals(jobId))
+                        .findFirst();
+        assertThat(job).isPresent();
+        assertThat(job.get().getStartTime()).isGreaterThan(currentTime);
+
+        // 2. verify the new job is a batch job
+        ArchivedExecutionGraph executionGraph =
```

Comment:
We can get the JobDetailsInfo using RestClusterClient, then we can use ObjectMapper the jobType, you can refer to PR: https://github.com/apache/flink/pull/24765/files,  MaterializedTableStatementITCase(L244). This may be one possible solution. I just think using ArchivedExecutionGraph is too lower-level API.